api/envoy/api/v2/endpoint/endpoint_components.proto

```diff
@@ -50,6 +54,11 @@ message Endpoint {
   //   This takes into effect only for upstream clusters with
   //   :ref:`active health checking <arch_overview_health_checking>` enabled.
   HealthCheckConfig health_check_config = 2;
+
+  // The hostname associated with this endpoint. This hostname is not used for routing or address
+  // resolution. If provided, it will be used in auto-host-rewrite routes and potentially health
+  // checks.
```

Comment:
instead of "potentially health checks", perhaps:
```suggestion
  // resolution. If provided, it will be used in auto-host-rewrite routes and health checks,
  // if configured.
```

The way it's written seems like we aren't certain about how the field will be used